Net31Solution/SensorApp/Sensor.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.ComponentModel;
+
+namespace SensorApp
+{
+    internal class Sensor : INotifyPropertyChanged
+    {
+        private int _measuredValue;
+        private Mode _sensorMode;
+
+        public int MeasuredValue
+        {
+            get
+            {
+                return _measuredValue;
+            }
+            set
+            {
+                _measuredValue = value;
+                RaisePropertyChanged(nameof(MeasuredValue));
+            }
+        }
+
+
+        public Guid Id { get; set; }
+        public int MeasurementInterval { get; set; }
+        public string SensorType { get; set; }
+        public Mode SensorMode
+        {
+            get
+            {
+                return _sensorMode;
+            }
+            set
+            {
+                _sensorMode = value;
+                RaisePropertyChanged(nameof(SensorMode));
+            }
+        }
+
+        public enum Mode
+        {
+            Idle,
+            Calibration,
+            Working
+        }
+        
+
+        public Sensor(ISensorConfig config)
+        {
+            Id = Guid.NewGuid();
+            MeasuredValue = config.MeasuredValue;
+            MeasurementInterval = config.MeasurementInterval;
+            SensorType = config.SensorType;
+            SensorMode = (Mode)config.SensorMode;
+        }
+
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public void RaisePropertyChanged(string propertyName)
+        {
+            PropertyChangedEventHandler temp = PropertyChanged;
+            if (temp != null)
+            {
+                temp(this, new PropertyChangedEventArgs(propertyName));
+            }
+        }
+
+        public void ChangeSensorMode()
+        {
+            if (SensorMode == Mode.Calibration)
+            {
+                SensorMode = Mode.Working;
+                GenerateWorkingValue();
```

Comment:
> Organize work with sensor modes using the State pattern. 

Drop Generate* method in class, reuse *ValueGenerator classes created previously. 
Create appropriate Generator based on mode. 
MeasuredValue getter should call generator each time when it is requested.